src/components/eventing/FilterForm.jsx

```diff
@@ -0,0 +1,139 @@
+
+const {Option} = Select;
+const { RangePicker } = DatePicker;
+
+const FilterForm = (props) => {
+  const dispatch = useDispatch();
+  const eventFilters = useSelector(state => state.uiState.eventFilters);
+  const triggerNames = useSelector(state => state.uiState.triggerNames);
+  
+  useEffect(() => {
+    dispatch(increment("pendingRequests")); 
+    client.eventing.fetchTriggerNames(props.projectID)
+    .then(res => {
+      const uniqueTriggerNames = [...new Set(res.map(val => val.rule_name))]
+      dispatch(set("uiState.triggerNames", uniqueTriggerNames))
+    })
+    .catch(ex => console.log(ex))
+    .finally(() => dispatch(decrement("pendingRequests")))
+  }, [])
+
+  const handleSubmit = e => {
+    e.preventDefault();
+    props.form.validateFields((err, fieldsValue) => {
+      let values = {};
+      if (!err) {
+        if(getFieldValue("showDate")){
+          const rangeValue = fieldsValue["range-picker"];
+          values = {
+            ...fieldsValue,
+            startDate: rangeValue[0].unix(),
+            endDate: rangeValue[1].unix()
+          }
+        }
+        else {
+          values = {...fieldsValue}
+        }
+        props.filterTable(values);
+        props.handleCancel();
+      }
+    });
+  }
+  const { getFieldDecorator, getFieldValue, resetFields } = props.form;
+
+  return (
+    <Modal
+      title="Filter event logs"
+      okText="Apply"
+      visible={props.visible}
+      onCancel={props.handleCancel}
+      onOk={handleSubmit}
+      cancelButtonProps={{style: {float: "left"}, onClick: () => {dispatch(reset("uiState.eventFilters"));props.handleCancel()}}}
+      cancelText="Reset Filters"
+    >
+      <Form layout="vertical" onSubmit={handleSubmit}>
+        <FormItemLabel name="Filter by status" />
+        <Form.Item>
+          {getFieldDecorator('status', {
+            initialValue: eventFilters.status ? eventFilters.status : ['processed', 'failed', 'staged']
+          })(
+            <Checkbox.Group>
+              <Checkbox value="processed">Processed <Icon type="check" style={{color: "#00FF00"}}/></Checkbox>
+              <Checkbox value="failed">Failed <Icon type="close" style={{color: "red"}}/></Checkbox>
+              <Checkbox value="staged">Pending <Icon type="hourglass" /></Checkbox>
+            </Checkbox.Group>
+          )}
+        </Form.Item>
+        <FormItemLabel name="Filter by trigger name" />
+        <Form.Item>
+          {getFieldDecorator('showName', {
+            initialValue: eventFilters.showName
+          })
+          (
+            <Checkbox 
+             checked={eventFilters.showName}
+             onChange={e => dispatch(set("uiState.eventFilters.showName", e.target.checked))}
+            >
+              Show logs of specific event triggers
+            </Checkbox>
+          )}
+        </Form.Item>
+        {getFieldValue('showName') && (
+          <>
+          <FormItemLabel name="Specify event triggers" />
+          <Form.Item>
```

Comment:
This is a multiple Select. Its value should be stored as an array in redux. Likewise, you should use `_in` clause in the graphql query for the trigger name.